python/whylogs/core/schema.py

```diff
@@ -14,6 +14,7 @@
 T = TypeVar("T", bound="DatasetSchema")
 
 
+@dataclass
```

Comment:
`@dataclass` or not?

Reply:
Yes I think data class is still appropriate. Boarderline, but this is def more of keeping state of a schema not a lot of logic